pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21148786?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21148786?buildTab=artifacts#/) on master @ [943f82fd320a79d23fc71ddb65f0223e10713081](https://github.com/cockroachdb/cockroach/commits/943f82fd320a79d23fc71ddb65f0223e10713081):


```
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:604
        	            	  | [...repeated from below...]
        	            	Wraps: (4) ***UNEXPECTED ERROR; Failed to generate a random operation
        	            	Wraps: (5) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*operationGenerator).randOp
        	            	  | 	pkg/workload/schemachange/operation_generator.go:220
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:591
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1268
        	            	Wraps: (6) failed generating operation: alterTableRLS
        	            	Wraps: (7) ERROR: relation "[150]" does not exist (SQLSTATE 42P01)
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.withPrefix (5) *withstack.withStack (6) *errutil.withPrefix (7) *pgconn.PgError
        	Test:       	TestWorkload
{
 "message": "Inspect Job Errors",
 "jobs": [
  {
   "jobId": 1152980556238258177,
   "status": "no errors reported"
  },
  {
   "jobId": 1152980295843840001,
   "status": "no errors reported"
  }
 ]
}
Schema Workload Stats
Total Schema Statements Executed = 107
Total Schema Statements Succeeded = 107
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 67
Total Transactions Rolled Back = 68
Total Transactions Executed = 135
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload805423400
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload805423400
--- FAIL: TestWorkload (172.06s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164106 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot T-sql-foundations branch-release-24.3.28-rc release-blocker]
- #163474 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [depends-on relation is dropped during DROP TABLE .. CASCADE] [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.4]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-60718

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21199348?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21199348?buildTab=artifacts#/) on master @ [d4edd299d4c98295fc768c46c813ecbce4bd7a27](https://github.com/cockroachdb/cockroach/commits/d4edd299d4c98295fc768c46c813ecbce4bd7a27):


```
    "result": false
   },
   {
    "query": "SELECT EXISTS (SELECT * FROM [SHOW REGIONS FROM DATABASE])",
    "result": false
   }
  ],
  "previousStatements": [
   "CREATE INDEX IF NOT EXISTS index_w0_87_w1_47 ON public.table_w0_87 (col87_w0_89 DESC, col87_w0_93 DESC, col87_w0_94 DESC, col87_w0_90 ASC, col87_w0_92, col87_w0_91 ASC) STORING (col87_w0_88)"
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:165: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:165
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3425
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:613
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1268
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 51
Total Schema Statements Succeeded = 51
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 33
Total Transactions Rolled Back = 42
Total Transactions Executed = 75
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1332497393
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1332497393
--- FAIL: TestWorkload (59.92s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164725 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-26.1]
- #164106 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3.28-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21201275?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21201275?buildTab=artifacts#/) on master @ [abe263a3c7f4a322a95f95e326c3fd9879fcc32a](https://github.com/cockroachdb/cockroach/commits/abe263a3c7f4a322a95f95e326c3fd9879fcc32a):


```
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3425
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:613
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1268
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
{
 "message": "Inspect Job Errors",
 "jobs": [
  {
   "jobId": 1155586514589450241,
   "status": "no errors reported"
  },
  {
   "jobId": 1155586282676027393,
   "status": "no errors reported"
  },
  {
   "jobId": 1155586088577236993,
   "status": "no errors reported"
  },
  {
   "jobId": 1155586050594996225,
   "status": "no errors reported"
  }
 ]
}
Schema Workload Stats
Total Schema Statements Executed = 157
Total Schema Statements Succeeded = 157
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 102
Total Transactions Rolled Back = 95
Total Transactions Executed = 197
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload609403081
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload609403081
--- FAIL: TestWorkload (159.98s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164725 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-26.1]
- #164106 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3.28-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21203719?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21203719?buildTab=artifacts#/) on master @ [fee209242c94fd6a3ba928dd7e0f2dd7c27f9058](https://github.com/cockroachdb/cockroach/commits/fee209242c94fd6a3ba928dd7e0f2dd7c27f9058):


```
     "public.table_w1_1"
    ],
    "result": [
     "lower(CAST(col1_w1_16 AS STRING))"
    ]
   }
  ],
  "previousStatements": [
   "ALTER TABLE public.table_w1_1 DROP COLUMN col1_w1_15"
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:165: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:165
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3425
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:613
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func7
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1268
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
Schema Workload Stats
Total Schema Statements Executed = 9
Total Schema Statements Succeeded = 9
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 6
Total Transactions Rolled Back = 9
Total Transactions Executed = 15
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1689506977
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1689506977
--- FAIL: TestWorkload (9.95s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164725 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-26.1]
- #164106 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3.28-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21206362?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21206362?buildTab=artifacts#/) on master @ [48c129d09165283d870911072443d3888cc6fceb](https://github.com/cockroachdb/cockroach/commits/48c129d09165283d870911072443d3888cc6fceb):


```
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3425
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:613
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func8
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1268
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
{
 "message": "Inspect Job Errors",
 "jobs": [
  {
   "jobId": 1155735294346002433,
   "status": "no errors reported"
  },
  {
   "jobId": 1155734136979881985,
   "status": "no errors reported"
  },
  {
   "jobId": 1155733931037491201,
   "status": "no errors reported"
  }
 ]
}
Schema Workload Stats
Total Schema Statements Executed = 158
Total Schema Statements Succeeded = 158
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 101
Total Transactions Rolled Back = 99
Total Transactions Executed = 200
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1076309683
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload1076309683
--- FAIL: TestWorkload (487.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164725 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-26.1]
- #164106 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3.28-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://mesolite.cluster.engflow.com/invocations/default/ffcde11d-0c36-4ab5-a463-2bcb3a938d90?testReportRun=2&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvY2NsL3Rlc3RjY2wvd29ya2xvYWQvc2NoZW1hY2hhbmdlOnNjaGVtYWNoYW5nZV90ZXN0) on master @ [3b93a56edfd7bea885cfd9f84f3d874c2d549971](https://github.com/cockroachdb/cockroach/commits/3b93a56edfd7bea885cfd9f84f3d874c2d549971):


```
    schema_change_external_test.go:165: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:165
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3425
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:613
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func8
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1693
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
{
 "message": "Inspect Job Errors",
 "jobs": [
  {
   "jobId": 1155765282126921729,
   "status": "no errors reported"
  },
  {
   "jobId": 1155765250225635329,
   "status": "no errors reported"
  }
 ]
}
Schema Workload Stats
Total Schema Statements Executed = 81
Total Schema Statements Succeeded = 81
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 54
Total Transactions Rolled Back = 43
Total Transactions Executed = 97
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/testlogs/pkg/ccl/testccl/workload/schemachange/schemachange_test/run_2_of_10/test.outputs/logTestWorkload175392896
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = on
    panic.go:615: -- test log scope end --
test logs left over in: outputs.zip/logTestWorkload175392896
--- FAIL: TestWorkload (42.10s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=2</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164725 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-26.1]
- #164106 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3.28-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/testccl/workload/schemachange/schemachange_test.TestWorkload [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21207630?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21207630?buildTab=artifacts#/) on master @ [5692c7894731454e1a1767a602577a3ad06922a6](https://github.com/cockroachdb/cockroach/commits/5692c7894731454e1a1767a602577a3ad06922a6):


```
  ],
  "usesDeclarativeSchemaChanger": true
 }
}
    schema_change_external_test.go:165: 
        	Error Trace:	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:165
        	Error:      	Received unexpected error:
        	            	***FAIL; Failed to receive an execution error when errors were expected
        	            	(1) forced error mark
        	            	  | "fatal error when running txn"
        	            	  | github.com/cockroachdb/errors/withstack/*withstack.withStack::
        	            	Wraps: (2)
        	            	Wraps: (3) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*opStmt).executeStmt
        	            	  | 	pkg/workload/schemachange/operation_generator.go:3425
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).runInTxn
        	            	  | 	pkg/workload/schemachange/schemachange.go:613
        	            	  | github.com/cockroachdb/cockroach/pkg/workload/schemachange.(*schemaChangeWorker).run
        	            	  | 	pkg/workload/schemachange/schemachange.go:701
        	            	  | pkg/ccl/testccl/workload/schemachange/schemachange_test_test.TestWorkload.TestWorkload.func5.func8
        	            	  | 	pkg/ccl/testccl/workload/schemachange/schema_change_external_test.go:154
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	external/org_golang_x_sync/errgroup/errgroup.go:93
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1268
        	            	Wraps: (4) ***FAIL; Failed to receive an execution error when errors were expected
        	            	Error types: (1) *markers.withMark (2) *schemachange.ErrorState (3) *withstack.withStack (4) *errutil.leafError
        	Test:       	TestWorkload
{
 "message": "Inspect Job Errors",
 "jobs": [
  {
   "jobId": 1155808398477361153,
   "status": "no errors reported"
  }
 ]
}
Schema Workload Stats
Total Schema Statements Executed = 157
Total Schema Statements Succeeded = 157
Total Schema Statement Expected Failures = 0
Total Transactions Committed = 108
Total Transactions Rolled Back = 91
Total Transactions Executed = 199
    schema_change_external_test.go:118: backup, tracing data, and system table dumps in /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2129572177
    schema_change_external_test.go:127: running DROP with use_declarative_schema_changer = off
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/67537770b84c3a4ea177539c1e46bd43/logTestWorkload2129572177
--- FAIL: TestWorkload (241.53s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #164725 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-26.1]
- #164106 pkg/ccl/testccl/workload/schemachange/schemachange_test: TestWorkload failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-24.3.28-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestWorkload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/workload/schemachange/operation_generator.go
+++ b/pkg/workload/schemachange/operation_generator.go
@@ -1069,7 +1069,6 @@ func (og *operationGenerator) createIndex(ctx context.Context, tx pgx.Tx) (*opSt
 	// If there are extra columns not used in the index, randomly use them
 	// as stored columns.
 	stmt := makeOpStmt(OpStmtDDL)
-	duplicateStore := false
 	isStoringVirtualComputed := false
 	regionColStored := false
 	columnNames = columnNames[len(def.Columns):]
@@ -1091,18 +1090,6 @@ func (og *operationGenerator) createIndex(ctx context.Context, tx pgx.Tx) (*opSt
 				}
 				if isVirtualComputed {
 					isStoringVirtualComputed = true
-				}
-			}
-
-			// If the column is already used in the primary key, then attempting to store
-			// it using an index will produce a pgcode.DuplicateColumn error.
-			if !duplicateStore {
-				colUsedInPrimaryIdx, err := og.colIsPrimaryKey(ctx, tx, tableName, columnNames[i].name)
-				if err != nil {
-					return nil, err
-				}
-				if colUsedInPrimaryIdx {
-					duplicateStore = true
 				}
 			}
 		}
@@ -1151,7 +1138,6 @@ func (og *operationGenerator) createIndex(ctx context.Context, tx pgx.Tx) (*opSt
 			// Inverted indexes cannot be unique.
 			{code: pgcode.InvalidSQLStatementName, condition: def.Unique && def.Type == idxtype.INVERTED},
 			{code: pgcode.InvalidSQLStatementName, condition: def.Type == idxtype.INVERTED && len(def.Storing) > 0},
-			{code: pgcode.DuplicateColumn, condition: duplicateStore},
 			{code: pgcode.FeatureNotSupported, condition: nonIndexableType},
 			{code: pgcode.FeatureNotSupported, condition: regionColStored},
 			{code: pgcode.FeatureNotSupported, condition: duplicateRegionColumn},

